### PR TITLE
CDAP-6643 Exponential backoff on exceptions in LogWriter

### PR DESCRIPTION
**Pull request by poornachandra**
JIRA - https://issues.cask.co/browse/CDAP-6643
Build - http://builds.cask.co/browse/CDAP-DUT4506-2

### CODE REVIEW

File: cdap-watchdog/src/main/java/co/cask/cdap/logging/save/ExponentialBackoff.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+package co.cask.cdap.logging.save;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Performs exponential backoff.
+ * Modified from Twill's SimpleKafkaConsumer class.
+ */
+public class ExponentialBackoff {
```
Comment (sagarkapare) [inline cdap-watchdog/src/main/java/co/cask/cdap/logging/save/ExponentialBackoff.java:26]: Should we move this class to cdap-common as the functionality exposed is is not specific to the logging?

**Comment by sagarkapare** [comment]
one question, otherwise LGTM

File: cdap-watchdog/src/main/java/co/cask/cdap/logging/save/ExponentialBackoff.java (hunk 1 above)
Comment (poornachandra) [inline cdap-watchdog/src/main/java/co/cask/cdap/logging/save/ExponentialBackoff.java:26]: I thought about it. We can do it later when there is another use for this class outside logging.

Comment (sagarkapare) [inline cdap-watchdog/src/main/java/co/cask/cdap/logging/save/ExponentialBackoff.java:26]: sounds good 👍

**Comment by poornachandra** [comment]
Build is green, merging.

**Pull request merged**